drivers/include/periph/i2c.h

```diff
@@ -0,0 +1,350 @@
+ * Some typical values are:<br>
+ * Normal mode:       10k&Omega;<br>
+ * Fast mode:          2k&Omega;<br>
+ * Fast plus mode:     2k&Omega;
+ *
+ * For more details refer to section 7.1 in:<br>
+ * http://www.nxp.com/documents/user_manual/UM10204.pdf
+ *
+ *
+ * @section   sec_pm (Low-) power implications
+ *
+ * The I2C interface realizes a transaction-based access scheme to the bus. From
+ * a power management perspective, we can leverage this by only powering on the
+ * I2C peripheral while it is actually used, that is inside an i2c_acquire() -
+ * i2c_release() block.
+ *
+ * After initialization, the I2C peripheral **should** be powered off (e.g.
+ * through peripheral clock gating). It should only be powered on once a
+ * transaction on the I2C bus starts, namely in the i2c_acquire() function. Once
+ * the transaction is finished, the corresponding I2C peripheral **should** be
+ * powered off again in the i2c_release() function.
+ *
+ * If the implementation puts the active thread to sleep while a transfer is in
+ * progress (e.g. when using DMA), the implementation might need to block
+ * certain power states.
+ *
+ * @{
+ * @file
+ * @brief       Low-level I2C peripheral driver interface definition
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Thomas Eichinger <thomas.eichinger@fu-berlin.de>
+ */
+
+#ifndef PERIPH_I2C_H
+#define PERIPH_I2C_H
+
+#include <stdint.h>
+#include <stddef.h>
+#include <limits.h>
+
+#include "periph_conf.h"
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Default I2C device access macro
+ * @{
+ */
+#ifndef I2C_DEV
+#define I2C_DEV(x)          (x)
+#endif
+/** @} */
+
+/**
+ * @brief   Default I2C undefined value
+ * @{
+ */
+#ifndef I2C_UNDEF
+#define I2C_UNDEF           (UINT_MAX)
+#endif
+/** @} */
+
+/**
+ * @brief   Default i2c_t type definition
+ * @{
+ */
+#ifndef HAVE_I2C_T
+typedef unsigned int i2c_t;
+#endif
+/**  @} */
+
+/**
+ * @brief   Read bit needs to be set when reading
+ */
+#define I2C_READ            (0x0001)
+
+/**
+ * @brief   Default mapping of I2C bus speed values
+ * @{
+ */
+#ifndef HAVE_I2C_SPEED_T
+typedef enum {
+    I2C_SPEED_LOW = 0,      /**< low speed mode:    ~10kbit/s */
+    I2C_SPEED_NORMAL,       /**< normal mode:       ~100kbit/s */
+    I2C_SPEED_FAST,         /**< fast mode:         ~400kbit/sj */
+    I2C_SPEED_FAST_PLUS,    /**< fast plus mode:    ~1Mbit/s */
+    I2C_SPEED_HIGH,         /**< high speed mode:   ~3.4Mbit/s */
+} i2c_speed_t;
+#endif
+/** @} */
+
+/**
+ * @brief   I2C transfer flags
+ * @{
+ */
+#ifndef HAVE_I2C_FLAGS_T
+typedef enum {
+    I2C_ADDR10  = 0x01,     /**< use 10-bit device addressing */
+    I2C_REG16   = 0x02,     /**< use 16-bit register addressing */
+    I2C_NOSTOP  = 0x04,     /**< do not issue a STOP condition after transfer */
+    I2C_NOSTART = 0x08      /**< skip START sequence, ignores address field */
```

Comment:
How do you do a repeated start? NOSTOP on the previous transaction followed by no flags on the next?